### ISSUE TITLE
Tabs content not rendered in SSR context

### ISSUE THREAD

**Issue by sylvaingi** [tailwindlabs/headlessui#1086]
**What package within Headless UI are you using?**

@headlessui/react

**What version of that package are you using?**

1.4.3

**What browser are you using?**

Chrome

**Reproduction URL**

https://github.com/sylvaingi/headless-ui-tabs-ssr-issue

**Describe your issue**

When using the Tab compoents in a SSR context (Next 12), the contents of the selected tab are not rendered.

For example the following code
```js
<Tab.Group>
  <Tab.List>
    <Tab>Tab 1</Tab>
    <Tab>Tab 2</Tab>
  </Tab.List>

  <Tab.Panels className="TAB_CONTENTS_HERE">
    <Tab.Panel>Contents of tab 1</Tab.Panel>
    <Tab.Panel>Contents of tab 2</Tab.Panel>
  </Tab.Panels>
</Tab.Group>
```

will yield the following HTML:
```html
<div role="tablist" aria-orientation="horizontal">
  <button
    id="headlessui-tabs-tab-undefined"
    role="tab"
    type="button"
    aria-selected="false"
    tabindex="-1"
  >
    Tab 1</button
  ><button
    id="headlessui-tabs-tab-undefined"
    role="tab"
    type="button"
    aria-selected="false"
    tabindex="-1"
  >
    Tab 2
  </button>
</div>
<div class="TAB_CONTENTS_HERE"></div>
```

**Comment by patrick-mcdougle** [comment]
I think the crux of the issue is that neither useEffect nor useLayoutEffect run on the server. There are quite a few of those that setup the initial states and such. This comment isn't too helpful to you the user, but it might be helpful to someone who would be implementing a fix.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏


This should be fixed by #1155, and will be available in the next release.

You can already try it using `npm install @headlessui/react@insiders` or `yarn add @headlessui/react@insiders`.

**Comment by JuHwon** [comment]
same here. layout shifts are gone, though it introduced hydration error:
`Warning: Did not expect server HTML to contain a <div> in <div>.`
